### PR TITLE
NCL-7430 Add OpenTelemetry instrumentation

### PR DESCRIPTION
**Pull request by rnc**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/bacon/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -81,6 +81,9 @@
         <maven.compiler.source>11</maven.compiler.source>
         <maven.compiler.target>11</maven.compiler.target>
 
+        <otel.alpha.version>1.19.0-alpha</otel.alpha.version>
```
Comment (vibe13) [inline pom.xml:84]: 1.19.2 is out

File: common/src/main/java/org/jboss/pnc/bacon/common/OTELHelper.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    public void stopOtel() {
+        if (root != null && processor != null) {
+            log.debug("Finishing OTEL instrumentation for {}", root);
+            root.end();
+            processor.close();
+        }
+    }
+
+    public String generateTraceParent(String trace, String span) {
+        return "00" + '-' + trace + '-' + span + "-01";
```
Comment (vibe13) [inline common/src/main/java/org/jboss/pnc/bacon/common/OTELHelper.java:77]: Ok this last "-01" might vary (depends mainly if the spans are sampled or not). I don't have a logic for that yet, but we might need to change that in the future, for now we can consider it ok!

**Comment by vibe13** [comment]
LGTM for what I can tell!

**Review by vibe13 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (rnc) [reply pom.xml:84]: Thanks! Actually I don't think I need that variable ( used it when testing the autoconfigurespi)

File: common/src/main/java/org/jboss/pnc/bacon/common/OTELHelper.java (hunk 2 above)
Comment (rnc) [reply common/src/main/java/org/jboss/pnc/bacon/common/OTELHelper.java:69]: Sure that makes sense.

**Review by vibe13 — APPROVED** [review]

**Pull request merged**